core/ecschema-editing/src/Editing/Formats.ts

```diff
@@ -6,65 +6,68 @@
+
+      if (units !== undefined) {
+        for (const unit of units) {
+          const unitItem = await this.getSchemaItem<Unit | InvertedUnit>(unit, null);
+
+          if (unitItem.schemaItemType !== SchemaItemType.Unit && unitItem.schemaItemType !== SchemaItemType.InvertedUnit)
+            throw new SchemaEditingError(ECEditingStatus.InvalidFormatUnitsSpecified, schemaItemIdentifier((unitItem as SchemaItem).schemaItemType, (unitItem as SchemaItem).key));
 
-        if (unitItem.schemaItemType !== SchemaItemType.Unit && unitItem.schemaItemType !== SchemaItemType.InvertedUnit)
-          return { errorMessage: `${(unitItem as SchemaItem).fullName} is not of type Unit or InvertedUnit.` };
-        newFormat.addUnit(unitItem);
+          newFormat.addUnit(unitItem);
+        }
       }
+    } catch (e: any) {
+      throw new SchemaEditingError(ECEditingStatus.CreateSchemaItemFailed, schemaItemIdentifierFromName(schemaKey, this.schemaItemType, name), e);
     }
+
     if (displayLabel)
       newFormat.setDisplayLabel(displayLabel);
 
     // TODO: Handle the setting of format traits, separators, etc....
     newFormat.setFormatType(formatType);
-    return { itemKey: newFormat.key };
+    return newFormat.key;
   }
 
   /**
    * Creates a format through a SchemaItemFormatProps.
    * @param schemaKey a SchemaKey of the Schema that will house the new object.
    * @param relationshipProps a json object that will be used to populate the new RelationshipClass. Needs a name value passed in.
    */
-  public async createFromProps(schemaKey: SchemaKey, formatProps: SchemaItemFormatProps): Promise<SchemaItemEditResults> {
-    const schema = await this._schemaEditor.getSchema(schemaKey);
-    if (schema === undefined)
-      return { errorMessage: `Schema Key ${schemaKey.toString(true)} not found in context` };
-
-    if (formatProps.name === undefined)
-      return { errorMessage: `No name was supplied within props.` };
-
-    const newFormat = (await schema.createFormat(formatProps.name)) as MutableFormat;
-    if (newFormat === undefined)
-      return { errorMessage: `Failed to create class ${formatProps.name} in schema ${schemaKey.toString(true)}.` };
+  public async createFromProps(schemaKey: SchemaKey, formatProps: SchemaItemFormatProps): Promise<SchemaItemKey> {
+    let newFormat: MutableFormat;
+    try {
+      const schema = await this.getSchema(schemaKey);
+      const boundCreate = schema.createFormat.bind(schema);
+      newFormat = await this.createSchemaItemFromProps<Format>(schemaKey, this.schemaItemType, boundCreate, formatProps) as MutableFormat;
+    } catch (e: any) {
+      throw new SchemaEditingError(ECEditingStatus.CreateSchemaItemFromPropsFailed, schemaItemIdentifierFromName(schemaKey, this.schemaItemType, formatProps.name!), e);
+    }
 
     await newFormat.fromJSON(formatProps);
-    return { itemKey: newFormat.key };
+    return newFormat.key;
```

Comment:
maybe we should capsule the whole json reading in the try block, otherwise if something fails on reading the props (line 70) it would not be wrapped with the SchemaEditingError.

Aside from that, we we actually need line 70? as far I understand the code, this.createSchemaItemsFromProps internally calls newItem.fromJSON already itself.